src/prefect/blocks/notifications.py

```diff
@@ -487,3 +489,135 @@ def block_initialization(self) -> None:
             ).url()
         )
         self._start_apprise_client(url)
+
+
+class SubstitutionHelper:
+    """A helper class for value substitution
+
+    It will be used to replace any `${key}` in strings inside nested data
+    with value in context.
+    """
+
+    def __init__(self, data: dict) -> None:
+        self.data = data
+
+    def get(self, m: re.Match):
+        key = m.group(1)
+        val = self.data[key]
+        if val is None:
+            return "null"
+        return str(val)
+
+    def handle(self, val: Any):
+        """Do substitution"""
+        if val is None:
+            return None
+        if isinstance(val, dict):
+            return {key: self.handle(subval) for key, subval in val.items()}
+        if isinstance(val, list):
+            return [self.handle(subval) for subval in val]
+        if isinstance(val, str):
+            return re.sub(r"\$\{(\w+)\}", self.get, val)
+        return val
+
+
+class CustomWebhookNotificationBlock(NotificationBlock):
+    """
+    Enables sending notifications via any custom webhook.
+
+    All nested string param contains `${key}` will be substituted with value from context/secrets.
+
+    Context values include: `subject`, `body` and `name`.
+
+    Examples:
+        Load a saved custom webhook and send a message:
+        ```python
+        from prefect.blocks.notifications import CustomWebhookNotificationBlock
+
+        custom_webhook_block = CustomWebhookNotificationBlock.load("BLOCK_NAME")
+
+        custom_webhook_block.notify("Hello from Prefect!")
+        ```
+    """
+
+    _block_type_name = "Custom Webhook"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/6ciCsTFsvUAiiIvTllMfOU/627e9513376ca457785118fbba6a858d/webhook_icon_138018.png?h=250"
+    _documentation_url = "https://docs.prefect.io/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.CustomWebhookNotificationBlock"
+
+    name: str = Field(title="Name", description="Name of the webhook.")
+
+    url: str = Field(
+        title="Webhook URL",
+        description="The webhook URL.",
+        example="https://hooks.slack.com/XXX",
+    )
+
+    method: Literal["GET", "POST", "PUT", "PATCH", "DELETE"] = Field(
+        default="POST", description="The webhook request method. Defaults to `POST`."
+    )
+
+    params: Optional[Dict[str, str]] = Field(
+        default=None, title="Query params", description="Custom query params."
+    )
+    json_data: Optional[dict] = Field(
+        default=None,
+        alias="json",
```

Comment:
I don't think this alias is necessary since `json_data` shouldn't collide with any Pydantic methods or attributes and you don't access `.json` anywhere as far as I see.

Reply:
It was an attempt to match the `httpx.request` method.
Guess it's more suitable to use `json_data` and `form_data` here, as I already named the title as `JSON Data` and `Form Data`.